### PR TITLE
[DPTP-3028] Resolve all input image tag namespace based on the runtime architecture

### PR DESCRIPTION
**Pull request by droslean**
Resolve the namespace based on the architecture. This PR shouldn't affect anything in production. 

/cc @bbguimaraes @openshift/test-platform 

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Review by bbguimaraes — CHANGES_REQUESTED** [review]
I fear this will become a continuous hunt for cases where the namespace should/shouldn't be altered, with the conflation of `ImageStreamTagReference` for internal and external images and the "stringly-typed" namespace fields.  What about `base_rpm_images`, `tag_specification`, `releases`, `build_root`, …?

To do this safely, we should IMO:

- derive a type from `ImageStreamTagReference` (possibly a corresponding one for `namespace` fields, as in e.g. `PromotionConfiguration`) for tags which need this special treatment
- go through the entirety of `pkg/api` / https://steps.ci.openshift.org/ci-operator-reference and change external references to use those types (`BaseImages`, `BaseRPMImages`, etc.)
- add the `NamespaceWithArch` method to _those types only_

We will also need tests.

File: pkg/util/util.go [hunk 1]
```diff
@@ -26,3 +28,11 @@ func SortedKeys[K constraints.Ordered, V any](m map[K]V) (ret []K) {
 	SortSlice(ret)
 	return
 }
+
+func NamespaceWithArch(namespace string) string {
+	arch := runtime.GOARCH
```
[the remark this answers is not in the document]
Comment (droslean) [reply pkg/util/util.go:33]: The only way to test it is to run the tests in `arm01` cluster.

File: pkg/api/types.go [hunk 2]
```diff
@@ -381,6 +382,14 @@ func (i *ImageStreamTagReference) ISTagName() string {
...
 }
 
+func (i *ImageStreamTagReference) ResolveNamespace() string {
+	arch := runtime.GOARCH
+	if arch == "amd64" {
+		return i.Namespace
+	}
+	return fmt.Sprintf("%s-%s", i.Namespace, arch)
+}
+
```
[the remark this answers is not in the document]
Comment (droslean) [reply pkg/api/types.go:392]: You can't even imagine what happens to the code if we introduce a new struct for this case.... :)

**Comment by droslean** [comment]
@bbguimaraes Yea we should do this one by one. I will introduce a new type for the multi-arch cases. In this PR I will try to change only `base_images` and `base_rpm_images` if it won't affect anything else...

**Comment by droslean** [comment]
@bbguimaraes The latest changes in this PR, allow all input images to resolve their namespace based on the architecture. I believe that this is a good way to start. I will follow up by adding this feature for the rest of the hardcoded images we are using like `clonerefs`

**Pull request merged**